src/services/email.service.js

```diff
@@ -0,0 +1,139 @@
+
+        const emails = input.replace(/\s+/g, '').split(',');
+
+        const invalidEmails = emails.filter(email => !emailRegex.test(email));
+
+        return invalidEmails.length <= 0;
+    }
+
+    static validateSelectedEmails(input) {
+        const invalidEmails = input.filter(obj => !this.validateEmails(obj.email));
+
+        return invalidEmails.length <= 0;
+    }
+
+    static async sendErrorEmail(rejectedEmail, message) {
+        await transporter.sendMail({
+            from: config.EMAIL_SEND_FROM_SMTP,
+            to: config.APP_ENV === 'production' ? config.EMAIL_SEND_FROM_SMTP : config.APP_DEV_EMAIL,
+            subject: 'Error: ' + rejectedEmail,
+            text: message,
+        });
+    }
+
+    static sendEmail(req) {
+        const {
+            mailTo,
+            cc,
+            bcc,
+            selectedEmails,
+            subject,
+            text,
+            emailFooter,
+            date,
+            time
+        } = req.body;
+
+        let selectedEmailsArray = [];
+
+        if (selectedEmails) {
+            selectedEmailsArray = JSON.parse(selectedEmails);
+        }
+
+        if (
+            !mailTo &&
+            !cc &&
+            !bcc &&
+            selectedEmailsArray.length === 0
+        ) {
+            throw new ValidationError('Nie podano żadnego adresu e-mail.');
+        } else if (
+            (!(this.validateEmails(mailTo)) && mailTo) ||
+            (!(this.validateEmails(cc)) && cc) ||
+            (!(this.validateEmails(bcc)) && bcc) ||
+            !this.validateSelectedEmails(selectedEmailsArray)
+        ) {
+            throw new ValidationError('Podano niepoprawny adres e-mail.');
+        }
+
+        let defaultEmails = [];
+        let ccEmails = [];
+        let bccEmails = [];
+
+        selectedEmailsArray.forEach((obj) => {
+            if (obj.method === 'default') {
+                defaultEmails.push(obj.email);
+            } else if (obj.method === 'cc') {
+                ccEmails.push(obj.email);
+            } else if (obj.method === 'bcc') {
+                bccEmails.push(obj.email);
+            }
+        });
+
+        const mailData = {
+            from: config.EMAIL_SEND_FROM_SMTP,
+            to: mailTo ? mailTo + ',' + defaultEmails.join(',') : defaultEmails.join(','),
+            cc: cc ? cc + ',' + ccEmails.join(',') : ccEmails.join(','),
+            bcc: bcc ? bcc + ',' + bccEmails.join(',') : bccEmails.join(','),
+            subject: subject,
+            text: text + '\n\n' + emailFooter + '\n',
+            // html: '',
+        };
+
+        if (req.file) {
+            mailData.attachments = [{path: req.file.path},];
+        }
+
+        const validateDateTime = (date, time) => {
+            const dateTime = new Date(`${date}T${time}`);
+            if (!date && !time) {
```

Comment:
You're right and the application logic is created to react when there is no date and time at the same time to create a new date object because you can send a message with no date and time as an option.  Otherwise, it will skip this `if` statement and communicate about the wrong date or time as a wrong input because as a default application guesses mistake from a user if there is only a date or time as an input.